test/e2e/apimachinery/garbage_collector.go

```diff
@@ -1101,14 +1101,14 @@ var _ = SIGDescribe("Garbage collector", func() {
 			framework.Failf("timeout in waiting for the owner to be deleted: %v", err)
 		}
 
-		// Wait 30s and ensure the dependent is not deleted.
-		ginkgo.By("wait for 30 seconds to see if the garbage collector mistakenly deletes the dependent crd")
-		if err := wait.PollWithContext(ctx, 5*time.Second, 30*time.Second+gcInformerResyncRetryTimeout, func(ctx context.Context) (bool, error) {
-			_, err := resourceClient.Get(ctx, dependentName, metav1.GetOptions{})
-			return false, err
-		}); err != nil && !wait.Interrupted(err) {
-			framework.Failf("failed to ensure the dependent is not deleted: %v", err)
-		}
+		// Wait 90s and ensure the dependent is not deleted.
+		ginkgo.By("wait for 90 seconds to see if the garbage collector mistakenly deletes the dependent crd")
+		gomega.Consistently(ctx, func(ctx context.Context) error {
+			_, err := framework.HandleRetry(func(ctx context.Context) (*unstructured.Unstructured, error) {
+				return resourceClient.Get(ctx, dependentName, metav1.GetOptions{})
+			})(ctx)
+			return err
```

Comment:
This can be simplified:
```
gomega.Consistently(ctx, framework.HandleRetry(func(ctx context.Context) (*unstructured.Unstructured, error) { return resourceClient.Get(ctx, dependentName, metav1.GetOptions{}))).WithTimeout(timeout).WithPolling(5 * time.Second).Should(gomega.Succeed())
```

Reply:
It seems that a match for error type is needed here？
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/121583/pull-kubernetes-e2e-kind/1719286645296992256